doc/src/sgml/high-availability.sgml

```diff
@@ -2972,11 +2969,10 @@ WALに記録された操作はすでにプライマリで発生したもので
     <command>VACUUM</command> against a table with no updated or deleted rows
     requiring cleanup might lead to conflicts.
 -->
-《機械翻訳》スタンバイクエリとの競合の原因として、行バージョンのクリーンアップだけではありません。
-すべてのインデックスのみのスキャン(スタンバイで実行されるものも含む)は、<acronym>MVCC</acronym>スナップショットを使用しなければなりません。
-これは、<quote>同意</quote>ビューと一致するものです。
-したがって、すべてのスタンバイクエリに対して<command>VACUUM</command>が1行以上の<emphasis>非</emphasis>可視性マップにある行を含むページを<command>VACUUM</command>で<link linkend="vacuum-for-visibility-map">設定</link>するときは、競合が必要になります。
-したがって、更新や削除された行を含まないテーブルに対して<command>VACUUM</command>を実行しても競合が発生する可能性があります。
+スタンバイクエリとの競合の潜在的な原因は、行バージョンのクリーンアップだけではありません。
```

Comment:
この前の文では「スタンバイクエリ」を「スタンバイ側の問い合わせ」
で「行バージョンのクリーンアップ」も「行バージョンの消去」にしてたようです。

Reply:
> この前の文では「スタンバイクエリ」を「スタンバイ側の問い合わせ」

スタンバイクエリ」は「スタンバイ側の問い合わせ」とします。

> 行バージョンのクリーンアップ」も「行バージョンの消去」にしてたようです。

"cleanup"を一律「消去」にして良いかどうか悩みます。
次の
「ですから、更新や削除された行を含まないテーブルに対して<command>VACUUM</command>を実行しても競合が発生する可能性があります。」は、実は”requiring cleanup"が約抜けしており、cleanupを「クリーンアップ」とすると、以下のようになりますが、

ですから、クリーンアップが必要で、更新や削除された行を含まないテーブルに対して<command>VACUUM</command>を実行しても競合が発生する可能性があります。

となりますが、cleanupを「消去」と訳すと、

ですから、消去が必要で、更新や削除された行を含まないテーブルに対して<command>VACUUM</command>を実行しても競合が発生する可能性があります。

となり、「消去」「削除」の日本語では似たような意味がある単語が同じ文の中に出てきて、かなりわかりにくくなるのではないかと思います。

ちなみに、high-availability.sgml の中で、cleanupをどう訳しているか抜き出したものが以下です。数字は行番号で、原文の後に訳が続いています。わかりやすくするために、[消去][クリーンアップ]でどちらの単語を選んでいるか表示しています。

集計すると、
[消去]: 6
[クリーンアップ]: 3

となり、多数に合わせるなら[消去]になりますが、上で述べたように、[消去]に意味が近い[削除]が同じ文で出現すると、わかりにくくなるという問題があります。個人的には、[クリーンアップ]としたほうが混乱が避けられると思いますが、いかがでしょう？

2800:         Application of a vacuum cleanup record from WAL conflicts with
WALからのバキュームクリーンアップレコードの適用は、その適用により削除される行のどれか1つでも<quote>見る</quote>ことができるスナップショットを持つスタンバイでのトランザクションとコンフリクトします。
[クリーンアップ]

2810:         Application of a vacuum cleanup record from WAL conflicts with
WALからのバキュームクリーンアップレコードは、消去されるデータが可視か否かに関係なく、スタンバイで対象ページにアクセスする問い合わせとコンフリクトします。
[クリーンアップ]

2946:    is <quote>early cleanup</quote>.  Normally, <productname>PostgreSQL</productname> allows
スタンバイ側の問い合わせとWAL再生の間でもっともよくあるコンフリクト理由は<quote>早すぎる消去</quote>です。
[消去]

2947:    cleanup of old row versions when there are no transactions that need to
通常<productname>PostgreSQL</productname>はMVCC規則にしたがって正確なデータの可視性を確実にするために、古い行バージョンを参照するトランザクションが存在しない場合それらを消去することが許されています。
[消去]

2950:    primary.  So it is possible that cleanup on the primary will remove row
したがって、スタンバイ上のトランザクションでまだ可視である行バージョンを、プライマリ上の消去処理が削除してしまう可能性があります。
[消去]

2961:    Row version cleanup isn't the only potential cause of conflicts with
スタンバイ問い合わせとの競合の潜在的な原因は、行バージョンのクリーンアップだけではありません。
[クリーンアップ]

2970:    requiring cleanup might lead to conflicts.
ですから、更新や削除された行を含まないテーブルに対して<command>VACUUM</command>を実行しても競合が発生する可能性があります。
cleanup訳漏れ

2996:    removing recently-dead rows and so cleanup conflicts do not occur.
これは<command>VACUUM</command>による最近不要になった行の削除を防止しますので、消去によるコンフリクトが発生しません。
[消去]

2998:    should note that this will delay cleanup of dead rows on the primary,
これを行う場合、プライマリで不要になった行の消去が遅延することに注意が必要です。望まないテーブルの膨張が発生してしまうかもしれません。
[消去]

2999:    which may result in undesirable table bloat. However, the cleanup
しかし、スタンバイ側で行うべき問い合わせをプライマリサーバ上で直接実行することと比べ、こうした消去に関する問題を優先する価値はありません。
[消去]

Reply:
そもそも、cleanupって、「消去」という意味があまり含まれていないのではないでしょうか。
参考：https://masahikosawada.github.io/2021/08/18/Btree-vacuum-improvements_in_PG14/

Reply:
詳細な調査ありがとうございます。
もともと統一されてないのが気になったので、
基本的にクリーンアップの方で良いように思えてきました。